src/main/java/com/salesforce/functions/jvm/sdk/bulk/BulkApi.java

```diff
@@ -0,0 +1,141 @@
+   * @throws BulkApiException If an error occurred while querying successful record results.
+   */
+  @SuppressWarnings("unused")
+  File getJobSuccessfulRecordResults(String jobId) throws BulkApiException;
+
+  /**
+   * Retrieves a list of failed records for a completed job.
+   *
+   * <p>The returned file is a CSV file that all the records that encountered an error while being
+   * processed by the job. Each row corresponds to a failed record.
+   *
+   * @see <a
+   *     href="https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/get_job_failed_results.htm">Bulk
+   *     API 2.0 and Bulk API Developer Guide</a>
+   * @param jobId The ID of the job to get failed record results for.
+   * @return A CSV file with all failed records.
+   * @throws BulkApiException If an error occurred while querying failed record results.
+   */
+  @SuppressWarnings("unused")
+  File getJobFailedRecordResults(String jobId) throws BulkApiException;
+
+  /**
+   * Uploads data for a job using CSV data you provide.
+   *
+   * @param jobId The ID for the job to upload data for.
+   * @param file The CSV file to upload to the job.
+   * @throws BulkApiException If an error occurred uploading the data.
+   * @throws IOException If an error occurred reading data from the provided file.
+   */
+  @SuppressWarnings("unused")
+  void uploadJobData(String jobId, File file) throws BulkApiException, IOException;
+
+  /**
+   * Uploads data for a job using CSV data you provide.
+   *
+   * @param jobId The ID for the job to upload data for.
+   * @param inputStream The CSV data as a stream to upload to the job.
+   * @throws BulkApiException If an error occurred uploading the data.
+   * @throws IOException If an error occurred reading data from the provided stream.
+   */
+  @SuppressWarnings("unused")
+  void uploadJobData(String jobId, InputStream inputStream) throws BulkApiException, IOException;
+
+  /**
+   * Uploads data for a job using CSV data you provide.
+   *
+   * @param jobId The ID for the job to upload data for.
+   * @param csv The CSV data to upload to the job.
+   * @throws BulkApiException If an error occurred uploading the data.
+   */
+  @SuppressWarnings("unused")
+  void uploadJobData(String jobId, String csv) throws BulkApiException;
+
+  /**
+   * Closes a job. If you close a job, Salesforce queues the job and uploaded data for processing,
+   * and you can’t add any more job data.
+   *
+   * @param jobId The ID of the job to close.
+   * @throws BulkApiException If an error occurred while closing the job.
+   */
+  @SuppressWarnings("unused")
+  void closeJob(String jobId) throws BulkApiException;
+
+  /**
+   * Aborts a job. If you abort a job, the job doesn't get queued or processed.
+   *
+   * @param jobId The ID of the job to abort.
+   * @throws BulkApiException If an error occurred while aborting the job.
+   */
+  @SuppressWarnings("unused")
+  void abortJob(String jobId) throws BulkApiException;
```

Comment:
The abort operations does return a more detailed response body.  Should that be represented in this API?

Reply:
I checked, we should be able to re-use `JobInfo` for this - added in [6865ae7](https://github.com/forcedotcom/sf-fx-sdk-java/pull/64/commits/6865ae7c5981059306e5de9a8e2a0feab0adcc09).